### PR TITLE
Support console scripts found in the PEX_PATH.

### LINKED ISSUE #1257: Support console scripts found in the PEX_PATH.
Right now, when building up a PEX with PEXBuilder, only the immediate distributions added to the PEXBuilder are searched for the console script entry point. Pants could use this to support a few binaries it assembles via PEX_PATH to avoid various invalidations.

### PR DESCRIPTION
**Pull request by jsirois**
Include the distributions found in any PEXes present on the PEX_PATH in
the search for console scripts at PEX build time.

Fixes #1257

### CODE REVIEW

File: pex/pex.py [hunk 1]
```diff
@@ -95,27 +95,38 @@ def interpreter(self):
         # type: () -> PythonInterpreter
         return self._interpreter
 
+    @property
```
Comment (Eric-Arellano) [inline pex/pex.py:98]: Thoughts on adding `@memoized_property` to Pex? This code will run twice iiuc.

File: tests/test_pex_builder.py [hunk 2]
```diff
@@ -266,3 +267,24 @@ def verify(pb):
             pb3.build(target)
             pb4 = build_from_req_pex(td4, target)
         verify(pb4)
+
+
+def test_prex_builder_script_from_pex_path(tmpdir):
```
Comment (Eric-Arellano) [inline tests/test_pex_builder.py:272]:
```suggestion
def test_pex_builder_script_from_pex_path(tmpdir):
```

**Review by Eric-Arellano — APPROVED** [review]

File: pex/pex.py (hunk 1 above)
Comment (jsirois) [reply pex/pex.py:98]: It caches manually in `self._envs` variable using the `if self._envs is None` guard.

File: tests/test_pex_builder.py (hunk 2 above)
Comment (jsirois) [reply tests/test_pex_builder.py:272]: Thanks - you know my answer there - I'll save some trees.

**Pull request merged**